docs/experiments-msmarco-passage.md

```diff
@@ -0,0 +1,115 @@
+
+Upon completion, we should have an index with 8,841,823 documents.
+The indexing speed may vary... on a modern desktop with an SSD, indexing takes a couple of minutes.
+
+## Retrieving and Evaluating the Dev set
+
+Since queries of the set are too many (+100k), it would take a long time to retrieve all of them. To speed this up, we use only the queries that are in the qrels file:
+
+```bash
+python tools/scripts/msmarco/filter_queries.py --qrels collections/msmarco-passage/qrels.dev.small.tsv \
+ --queries collections/msmarco-passage/queries.dev.tsv --output collections/msmarco-passage/queries.dev.small.tsv
+```
+
+The output queries file should contain 6980 lines.
+
+We can now retrieve this smaller set of queries:
+
+```bash
+python tools/scripts/msmarco/retrieve.py --hits 1000 --threads 1 \
+ --index indexes/msmarco-passage/lucene-index-msmarco --queries collections/msmarco-passage/queries.dev.small.tsv \
+ --output runs/run.msmarco-passage.dev.small.tsv
+```
+
+Note that by default, the above script uses BM25 with tuned parameters `k1=0.82`, `b=0.68`.
+The option `-hits` specifies the of documents per query to be retrieved.
+Thus, the output file should have approximately 6980 * 1000 = 6.9M lines.
+
+Retrieval speed will vary by machine:
+On a modern desktop with an SSD, we can get ~0.07 s/query, so the run should finish in under ten minutes.
+We can also perform multi-threaded retrieval by changing the `--threads` argument.
+Similarly, we can perform multi-threaded retrieval by changing the `-threads` argument.
+
+Finally, we can evaluate the retrieved documents using this the official MS MARCO evaluation script:
+
+```bash
+python tools/scripts/msmarco/msmarco_eval.py \
+ collections/msmarco-passage/qrels.dev.small.tsv runs/run.msmarco-passage.dev.small.tsv
+```
+
+And the output should be like this:
+
+```
+#####################
+MRR @10: 0.18741227770955546
+QueriesRanked: 6980
+#####################
+```
+
+We can also use the official TREC evaluation tool, `trec_eval`, to compute other metrics than MRR@10. 
+For that we first need to convert runs and qrels files to the TREC format:
+
+```bash
+python tools/scripts/msmarco/convert_msmarco_to_trec_run.py \
+ --input runs/run.msmarco-passage.dev.small.tsv \
+ --output runs/run.msmarco-passage.dev.small.trec
+
+python tools/scripts/msmarco/convert_msmarco_to_trec_qrels.py \
+ --input collections/msmarco-passage/qrels.dev.small.tsv \
+ --output collections/msmarco-passage/qrels.dev.small.trec
+```
+
+And run the `trec_eval` tool:
+
+```bash
+tools/eval/trec_eval.9.0.4/trec_eval -c -mrecall.1000 -mmap \
+ collections/msmarco-passage/qrels.dev.small.trec runs/run.msmarco-passage.dev.small.trec
+```
+
+The output should be:
+
+```
+map                   	all	0.1957
+recall_1000           	all	0.8573
+```
+
+Average precision and recall@1000 are the two metrics we care about the most.
```

Comment:
Maybe an explanation of why we care about these instead of MRR @ 10 which is their official metric? (i.e. something about multi-stage ranking)

Reply:
Will do!